### ISSUE TITLE
[Accessibility]Narrator cannot announce localized control type of TextBox/MaskedTextBox

### ISSUE THREAD

**Issue by Marie-Shi** [dotnet/winforms#2933]
* .NET Core Version: 
.Net Core 5.0.100-preview.2.20151.5 from 5.0 master branch + VS2019 with latest core designer

* Have you experienced this same bug with .NET Framework?: 
Yes

**More Info:**
1.	This issue can reproduce on TextBox/MaskedTextBox/TextBox with MultiLine=true.
2.	This issue can also reproduce on .Net Core 3.1 release branch.

**Steps to reproduce:**
1.	Open the attached app “Narrator_Not_Annouce_control_type.zip” in VS.
[Narrator_Not_Annouce_control_type.zip](https://github.com/dotnet/winforms/files/4273713/Narrator_Not_Annouce_control_type.zip)

2.	Build and run this application. 
3.	Start Narrator and focus on TextBox, observe Narrator behavior.

**Actual behavior:**
Narrator cannot announce localized control type of TextBox, it just announces “edit”, missing “TextBox”.
![Actual](https://user-images.githubusercontent.com/45864017/75650624-6738a400-5c91-11ea-9e7e-83b9848fb1b7.png)

**Expected behavior:**
Narrator should announce localized control type of TextBox. The behavior should be like below:
![Expected](https://user-images.githubusercontent.com/45864017/75650725-c0a0d300-5c91-11ea-94dd-b9d535de8d38.png)

**Comment by merriemcgaw** [comment]
@Marie-Shi is this only with multi-line set to True?

**Comment by Marie-Shi** [comment]
@merriemcgaw 
Question1: For TextBox with MultiLine=true, it just means set multi-line as true for TextBox.
Question2: This issue cannot reproduce on accessibility sample apps, because we add a label control before TextBox, so Narrator can announce TextBox correctly. For our accessibility tesing, it is needed to add a label before editing control. Below is the results:
Add label before editing control:
![111](https://user-images.githubusercontent.com/45864017/75953319-11ac0380-5eec-11ea-91c7-cb18d8f79681.png)

No label before editing control:
![222](https://user-images.githubusercontent.com/45864017/75953430-4c15a080-5eec-11ea-8391-7841a55d84c3.png)
Whether we need to close this GitHub issue?

**Comment by merriemcgaw** [comment]
I see. That's interesting - we are accessible as long as there is a control before us. I think this hasn't been blocking most teams recently as we haven't seen reports of it. @M-Lipin @vladimir-krestov  - we saw something like this with the MonthCalendar control, right? Did we fix it?

**Comment by M-Lipin** [comment]
To me the fix should change the localized control type but not the accessible name. So the result should be as follows:
```
First name edit -> First name edit
```
```
Address edit -> Address multi line edit
```
Where `First name` and `Address` are the accessible names defined by app developer and `edit` and `multi line edit` are locilizable control type which should be changed in the scope of this bug.

MonthCalendar control had the issue with defining accessible name by label. But this may be already fixed. Need to double check.

**Comment by M-Lipin** [comment]
@merriemcgaw could you please share your opinion about the following localizable control type names:
`edit` for the TextBox (no changes)
`multi line edit` for the TextBox with multiline=true
`masked edit` for MaskedTextBox

**Comment by merriemcgaw** [comment]
Those sound just fine to me. @guybark are you cool with these?

**Comment by guybark** [comment]
I think it's great to be considering the most helpful customer experience here, but I'd be a little wary around what would mean with respect to presenting a consistent experience. Will any other UI framework provide a localized control type of "multi line edit"? (I expect UIA provides a LocalizedControlType of "edit" in English if no LocalizedControlType is supplied by the provider.) If no other UI framework supplies "multi line edit", then when reaching a non-WinForms multi-line edit control and hearing "edit", will the customer start assuming that it's not multi-line? (And that could happen in the same app if it uses a mix of WinForms and WPF UI.) And will a different screen reader that doesn't announce the UIA LocalizedControlType property announce the multiline edit control as "edit"? Personally, I think it'd be good to discuss the suggested change with other parties, (eg WPF, UIA,) to keep a consistent experience. (Also, would most customers know what a "masked edit" control is? It sounds a bit like a password field where the characters being typed are obscured.)

**Comment by merriemcgaw** [comment]
Thanks @guybark! @fabiant3-zz, who would you suggest we work with to get consistent on the accessibility of localized control type?

**Comment by SergeySmirnov-Akvelon** [comment]
Hi @merriemcgaw. I have investigated behavior of WPF for this case. Narrator announce following messages for TextBox and RichTextBox:

- TextBox with AccessibleName:
`{FormTitle} window, {ControlAccessibleName}, edit {value},` (for example: `Test Application window, Test name, edit, Test text,`)
- TextBox withoit AccessibleName
`{FormTitle} window, edit {value},` (for example: `Test Application window, edit, Test text,`)
- RichTextBox with AccessibleName
`{FormTitle} window, {ControlAccessibleName}, editing,` (for example: `Test Application window, Test name, editing,`)
- RichTextBox without AccessibleName
`{FormTitle} window, editing,` (for example: `Test Application window, editing,`)

**Comment by merriemcgaw** [comment]
I don't see any violations here that we have to address. If we start to implement UIA ourselves for RichTextBox we lose everything that the control natively provides and have to do all that work ourselves. Unless there's a violation here that must be addressed I'm not inclined to do that work. In fact, I would try to get any violation fixed by the RichEdit control itself first.  Closing this issue.
